pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -77,6 +77,7 @@ type PodGCStrategy string
 const (
 	PodGCOnPodCompletion      PodGCStrategy = "OnPodCompletion"
 	PodGCOnPodSuccess         PodGCStrategy = "OnPodSuccess"
+	PodGCOnOnPodLabelSelected PodGCStrategy = "OnPodLabelSelected"
```

Comment:
do we need this, I think you could use OnPodCompletion with labels

Reply:
One use case is that we'd like to move pods to GC queue no matter what phase they are in, e.g. they might be running but marked with something that indicates that they are evicted and need to be deleted.

I just realized that I need to move the logic to check labels to be above the `if node.Fulfilled() && !node.IsDaemoned()` block to support that use case correctly.

What do you think?

Reply:
Actually we can ignore the case where pods are running for now and focus on filtering for completed pods.

I just switched to reuse OnPodCompletion with labels. Please take another look when you get a chance.